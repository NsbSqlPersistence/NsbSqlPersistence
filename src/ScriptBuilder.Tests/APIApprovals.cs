using NUnit.Framework;
using PublicApiGenerator;
using System.IO;
using System.Reflection;
using System.Runtime.CompilerServices;

[TestFixture]
public class APIApprovals
{
    [Test]
    [MethodImpl(MethodImplOptions.NoInlining)]
    public void Approve()
    {
        var combine = Path.Combine(TestContext.CurrentContext.TestDirectory, "NServiceBus.Persistence.Sql.ScriptBuilder.dll");
        var assembly = Assembly.LoadFile(combine);
        var publicApi = ApiGenerator.GeneratePublicApi(assembly, excludeAttributes: new[] { "System.Runtime.Versioning.TargetFrameworkAttribute" });
        TestApprover.Verify(publicApi);
    }
}