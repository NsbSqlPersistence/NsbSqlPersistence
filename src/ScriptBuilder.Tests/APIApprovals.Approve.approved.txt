namespace NServiceBus.Persistence.Sql
{
    public class ErrorsException : System.Exception
    {
        public ErrorsException(string message) { }
        public string FileName { get; set; }
    }
}
namespace NServiceBus.Persistence.Sql.ScriptBuilder
{
    public enum BuildSqlVariant
    {
        MsSqlServer = 0,
        MySql = 1,
    }
    public class CorrelationProperty
    {
        public CorrelationProperty(string name, NServiceBus.Persistence.Sql.ScriptBuilder.CorrelationPropertyType type) { }
        public string Name { get; }
        public NServiceBus.Persistence.Sql.ScriptBuilder.CorrelationPropertyType Type { get; }
    }
    public enum CorrelationPropertyType
    {
        String = 0,
        DateTime = 1,
        DateTimeOffset = 2,
        Int = 3,
        Guid = 4,
    }
    public class static MsSqlServerCorrelationPropertyTypeConverter
    {
        public static string GetColumnType(NServiceBus.Persistence.Sql.ScriptBuilder.CorrelationPropertyType propertyType) { }
    }
    public class static MySqlCorrelationPropertyTypeConverter
    {
        public static string GetColumnType(NServiceBus.Persistence.Sql.ScriptBuilder.CorrelationPropertyType propertyType) { }
    }
    public class static OutboxScriptBuilder
    {
        public static void BuildCreateScript(System.IO.TextWriter writer, NServiceBus.Persistence.Sql.ScriptBuilder.BuildSqlVariant sqlVariant) { }
        public static string BuildCreateScript(NServiceBus.Persistence.Sql.ScriptBuilder.BuildSqlVariant sqlVariant) { }
        public static void BuildDropScript(System.IO.TextWriter writer, NServiceBus.Persistence.Sql.ScriptBuilder.BuildSqlVariant sqlVariant) { }
        public static string BuildDropScript(NServiceBus.Persistence.Sql.ScriptBuilder.BuildSqlVariant sqlVariant) { }
    }
    public class SagaDefinition
    {
        public SagaDefinition(string tableSuffix, string name, NServiceBus.Persistence.Sql.ScriptBuilder.CorrelationProperty correlationProperty, NServiceBus.Persistence.Sql.ScriptBuilder.CorrelationProperty transitionalCorrelationProperty = null) { }
        public SagaDefinition(string tableSuffix, string name) { }
        public NServiceBus.Persistence.Sql.ScriptBuilder.CorrelationProperty CorrelationProperty { get; }
        public bool IsAlwaysStartNew { get; }
        public string Name { get; }
        public string TableSuffix { get; }
        public NServiceBus.Persistence.Sql.ScriptBuilder.CorrelationProperty TransitionalCorrelationProperty { get; }
    }
    public class static SagaScriptBuilder
    {
        public static string BuildCreateScript(NServiceBus.Persistence.Sql.ScriptBuilder.SagaDefinition saga, NServiceBus.Persistence.Sql.ScriptBuilder.BuildSqlVariant sqlVariant) { }
        public static void BuildCreateScript(NServiceBus.Persistence.Sql.ScriptBuilder.SagaDefinition saga, NServiceBus.Persistence.Sql.ScriptBuilder.BuildSqlVariant sqlVariant, System.IO.TextWriter writer) { }
        public static void BuildDropScript(NServiceBus.Persistence.Sql.ScriptBuilder.SagaDefinition saga, NServiceBus.Persistence.Sql.ScriptBuilder.BuildSqlVariant sqlVariant, System.IO.TextWriter writer) { }
        public static string BuildDropScript(NServiceBus.Persistence.Sql.ScriptBuilder.SagaDefinition saga, NServiceBus.Persistence.Sql.ScriptBuilder.BuildSqlVariant sqlVariant) { }
    }
    public class static SubscriptionScriptBuilder
    {
        public static void BuildCreateScript(System.IO.TextWriter writer, NServiceBus.Persistence.Sql.ScriptBuilder.BuildSqlVariant sqlVariant) { }
        public static string BuildCreateScript(NServiceBus.Persistence.Sql.ScriptBuilder.BuildSqlVariant sqlVariant) { }
        public static void BuildDropScript(System.IO.TextWriter writer, NServiceBus.Persistence.Sql.ScriptBuilder.BuildSqlVariant sqlVariant) { }
        public static string BuildDropScript(NServiceBus.Persistence.Sql.ScriptBuilder.BuildSqlVariant sqlVariant) { }
    }
    public class static TimeoutScriptBuilder
    {
        public static void BuildCreateScript(System.IO.TextWriter writer, NServiceBus.Persistence.Sql.ScriptBuilder.BuildSqlVariant sqlVariant) { }
        public static string BuildCreateScript(NServiceBus.Persistence.Sql.ScriptBuilder.BuildSqlVariant sqlVariant) { }
        public static void BuildDropScript(System.IO.TextWriter writer, NServiceBus.Persistence.Sql.ScriptBuilder.BuildSqlVariant sqlVariant) { }
        public static string BuildDropScript(NServiceBus.Persistence.Sql.ScriptBuilder.BuildSqlVariant sqlVariant) { }
    }
}